### PR TITLE
Turn this fork into an installable package

### PR DESCRIPTION
**Pull request by honzajavorek**
- Apply our internal changes to the `bin/uritemplate.js` file
- Since one of the changes is that the UriTemplate objects are not frozen anymore, remove the integration test checking exactly that
- Remove everything related to the source of the library, linter, unit tests, etc. (we modified the result of compilation, so either we would have to backport the changes to the individual parts, or we need to keep the modified result of compilation as a single source of truth)
- Modify `Jakefile.js` so it only run integration tests for the library, add Circle CI integration
- Replace `const` with `var` in our changes
- Change `package.json` in order to be able to publish the fork to our internal npm registry

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,55 +1,28 @@
...
       "jake": "latest",
-      "uglify-js": "latest",
       "async": "latest"
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/fxa/uritemplate-js.git"
+    "url": "https://github.com/apiaryio/uritemplate-js.git"
+  },
+  "publishConfig": {
```
Comment (kuba-kubula) [inline package.json:25]: you don't need this, as long as your terminal has .npmrc with configuration for `@apiaryio/...` package name

**Review by kuba-kubula — APPROVED** [review]
@honzajavorek Just one minor nitpick, but looks ok.

**Comment by kuba-kubula** [comment]
@honzajavorek also - I'd prefer moving the `bin/uritemplate.js` into `lib` folder, but... /shrug as it would make comparing master<->upstream impossible.

File: package.json (hunk 1 above)
Comment (honzajavorek) [reply package.json:25]: It's a good practice against mistakes like cloning the repo and typing `npm publish` and accidentally publishing to the public registry or sth like that. We have that in multiple internal projects.

**Comment by honzajavorek** [comment]
@kuba-kubula Yup, I tried to only delete things, so my commit is easily disposable. Moving things around would make upstream updates harder.

**Comment by honzajavorek** [comment]
@kuba-kubula Thanks! 💛 💙

File: package.json (hunk 1 above)
Comment (kuba-kubula) [reply package.json:25]: ok then.

**Pull request merged**